### PR TITLE
Delete nodes from k8s api during rolling-update

### PR DESCRIPTION
**Pull request by justinsb**
This prevents a race where if the new node comes back with the same
name, it will still be cordoned.  This seems to be more likely on GCE.

### CODE REVIEW

File: pkg/instancegroups/instancegroups.go [hunk 1]
```diff
@@ -172,12 +175,26 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 			}
 		}
 
+		// We unregister the node before deleting it; if the replacement comes up with the same name it would otherwise still be cordoned
+		// (GCE seems to recycle names more rapidly)
+		if !isBastion && !rollingUpdateData.CloudOnly {
+			if u.Node == nil {
+				glog.Warningf("skipping node unregister from kubernetes; node was not set")
```
[the remark this answers is not in the document]
Comment (justinsb) [reply pkg/instancegroups/instancegroups.go]: Switched to ` glog.Warningf("no kubernetes Node associated with %s, skipping node deletion", instanceId)`

File: pkg/instancegroups/instancegroups.go [hunk 2]
```diff
@@ -172,12 +175,26 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 			}
 		}
 
+		// We unregister the node before deleting it; if the replacement comes up with the same name it would otherwise still be cordoned
+		// (GCE seems to recycle names more rapidly)
```
[the remark this answers is not in the document]
Comment (justinsb) [reply pkg/instancegroups/instancegroups.go]: Done!

**Comment by justinsb** [comment]
So I think it's a kube problem, but the role of kops is just to make sure that we don't have two instances with the same name.

It's much less likely on AWS, where we name instances after their internal IP, but it could happen if the new instance reuses the same IP.

It seems to _always_ happen on GCE with MIGs, at least right now.  Not sure whether that has always been the case.

I guess the right place to fix it is for the kubelet to know that it isn't the same node as the existing node, i.e. kubelet should check the machineid or something.  But that isn't happening as far as I can tell (although I think I was testing with CoreOS, and maybe CoreOS has messed up whatever check kubelet is using).

Anyway, I think deleting the node feels like the safe thing to do - it's just a little more predictable.

**Comment by geojaz** [comment]
/lgtm

Ok cool- that makes sense to me, thanks for the fix and the info!

**Pull request merged**